src/py42/clients/util.py

```diff
@@ -2,12 +2,13 @@
 
 
 def get_all_pages(func, key, *args, **kwargs):
-
-    item_count = settings.items_per_page
     page_num = 0
-    while item_count >= settings.items_per_page:
+    if kwargs.get("page_size") is None:
```

Comment:
```python
page_size = kwargs.get("page_size") or settings.items_per_page
item_count = page_size
while item_count >= page_size:
   ...
```

Reply:
The problem is the page size is already part of kwargs and then it ends up getting passed twice to the function below

Reply:
Tests start failing when I make that change - I would have to re-architect other stuff to make that work. Is it ok the way it is?

Reply:
I even tried with line 11 back to the way it was

Reply:
it's fine the way it is.